giddy/util.py

```diff
@@ -81,3 +81,85 @@ def get_lower(matrix):
     return lowvec
 
 
+def fill_diag2(p):
```

Comment:
This might be clearer implemented as:

```python
def fill_empty_diagonals(p):
    if len(shape) == 3: 
         return _fill_empty_diagonal_3d(p)
    elif len(shape) == 2:
        return _fill_empty_diagonal_2d(p)
    else:
        raise NotImplementedError('filling empty diagonals is only implemented for 2/3d matrices.')
```

Reply:
I was struggling about which strategy to adopt when designing the interface. Thanks for the advice!

Reply:
resolved in https://github.com/pysal/giddy/pull/109/commits/0676528b415553dc043ea33b02bc89600d5edba9